plugins/services/tasks/local.go

```diff
@@ -62,8 +62,14 @@ import (
 )
 
 var (
-	_     = (api.TasksClient)(&local{})
-	empty = &ptypes.Empty{}
+	_                    = (api.TasksClient)(&local{})
+	empty                = &ptypes.Empty{}
+	tasksServiceRequires = []plugin.Type{
+		plugins.EventPlugin,
+		plugins.RuntimePluginV2,
+		plugins.MetadataPlugin,
+		plugins.TaskMonitorPlugin,
```

Comment:
This is a much more interesting case. If I read the code properly, TaskMonitorPlugin only exists on Linux. But things work _somehow_ on other OSes nevertheless.

Reply:
Same as above , this is just about generating the graph for proper plugin init ordering. The error would happen if the plugin's init attempted to get that plugin.